src/hooks/core/chat/useFetchChannelHook.ts

```diff
@@ -77,23 +92,56 @@ const useFetchChannelHook = () => {
 
     try {
       const members = getChannelMembers(channel);
-      await Promise.all(
-        (members || []).map(async (member) => {
-          const userMember = UserSchema.fromMemberWebsocketObject(member, channel?.id);
-          await userMember.saveOrUpdateIfExists(localDb);
-        })
-      );
-
-      await Promise.all(
-        (channel?.messages || []).map(async (message) => {
-          if (message?.type === 'deleted') return;
-          if (message?.type === 'system') {
-            message = getFirstMessage([message]);
+
+      (members || []).map((member) => {
+        const userMember = UserSchema.fromMemberWebsocketObject(member, channel?.id);
+        queue.addJob({
+          label: `saveUserMember-${member?.user?.id}`,
+          task: () => {
+            return new Promise((resolve) => {
+              UserSchema.isUserExists(localDb, member?.user?.id, channel?.id).then(
+                (isUserExists) => {
+                  if (!member?.user?.username) return resolve(true);
+                  if (!isUserExists) {
+                    userMember.save(localDb).then(() => {
+                      return resolve(true);
+                    });
+                  } else {
+                    return resolve(true);
+                  }
+                }
+              );
+            });
+          }
+        });
+
+        return null;
+      });
+
+      (channel?.messages || []).map((message) => {
+        queue.addJob({
+          label: `saveChat-${message?.id}`,
+          task: () => {
+            return new Promise((resolve) => {
+              if (message?.type === 'deleted') {
+                resolve(true);
+                return;
+              }
+              if (message?.type === 'system') {
+                message = getFirstMessage([message]);
+              }
+              const chat = ChatSchema.fromGetAllChannelAPI(channel?.id, message);
+              chat.saveIfNotExist(localDb).then(() => {
+                resolve(true);
+
+                refreshWithId('chat', channel?.id);
+              });
+            });
           }
-          const chat = ChatSchema.fromGetAllChannelAPI(channel?.id, message);
-          await chat.save(localDb);
-        })
-      );
+        });
+
+        return null;
+      });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made to this section are also beneficial for performance. Instead of using `await` inside a `map` function which could potentially block the execution, the new code adds jobs to the queue which will be executed asynchronously. This change should improve the overall performance of the application.



<!-- This is an auto-generated comment by OSS CodeRabbit -->